### DISCUSSION TITLE
SSG page data json files respond with rendered page html

### DISCUSSION THREAD

**Discussion by gijsbotje** [vercel/next.js#39505] [Help]
In my latest deployments of https://dbm-nextjs.vercel.app/ all SSG pages work on first load, but client side routing breaks. This is because for these pages, the JSON file that should contain the pageProps, somehow return the rendered page as HTML.

For example: https://dbm-nextjs.vercel.app/blog renders fine
but when you click on a blog and navigate back, you'll end up on a page with only the layout and no data.
It should load the pageProps from https://dbm-nextjs.vercel.app/_next/data/p68BYzThr7N3EAjTTvZgA/nl-nl/blog.json, but this returns HTML.

How is this possible? 6 days ago it worked fine: https://dbm-nextjs-n7ru7lzx6-afosto.vercel.app/_next/data/l3byLPuS4BLk5D5N0cHjS/nl-nl/blog.json
now it doesn't: https://dbm-nextjs.vercel.app/_next/data/p68BYzThr7N3EAjTTvZgA/nl-nl/blog.json

Running a local build with `yarn start` works fine. It's only the deployments online that are broken.

**Comment by ijjk** [comment]
Hi, can you share a repo with a minimal reproduction with this? Have there been any changes in your middleware file between those deployments?

**Comment by 3pns** [comment]
I got a similar problem except that I use no getStatic or GetServerSide props methods but next.js still try to fetch the .json files for each page. On development it returns an HTML page and using a static export on production it will attempt to download the .json which will cause 404 errors.

**Comment by lipoolock** [comment]
Hello @ijjk 
I have been faced the same issue on my side
I crafted a codesandbox to expose you more or less this phenomenon

https://codesandbox.io/s/e0eu71

If you launch the server an go to this URL for instance https://e0eu71.sse.codesandbox.io/1/a you will see some data (given by a public API)
All right !
But if you try to reach this URL https://e0eu71.sse.codesandbox.io/_next/data/development/1/a.json it will render the same page. That's a problem I guess 

In this codesandbox you can remove the `.bak` for `middleware.js.bak` and restart the server, then you will see as a first call the URL provided just above exposed in Network panel of your browser

Thanks for your feedback ;)

**Reply by lipoolock** [reply]
Hi @ijjk 
I updated to latest `12.3.0` stable release and this behavior seems to be still there.
Do you mind having a look at this issue please ?

**Reply by Eloujah** [reply]
It is still the case with i18n and next@14.1.0 (including canary)

**Comment by gijsbotje** [comment]
I'm not sure how and why, but for me the issues were resolved. Haven't updated the project to 14.1, so I hope it doesn't return for me when I upgrade it.
Not going to close the discussion yet, as @Eloujah mentioned that he encountered it in 14.1.
